src/runtime/recipe/lib-recipe.ts

```diff
@@ -126,6 +126,7 @@ export interface Handle {
   joinDataFromHandle(handle: Handle): void;
 
   findConnectionByDirection(direction: Direction): HandleConnection;
+  restrictType(restrictedType: Type): void;
```

Comment:
I wish we had documentation on these methods, but I assume that will come as part of step (3) @shans has planned re: reviewing/defining what the interface we really want will look like.